.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+default_language_version:
+  python: python3
+
+# NOTE: Update in pyproject.toml as well
+exclude: |
+  (?x)^(
+    __pycache__|
+    .+\/.+\/migrations\/.*|
+    \.venv
+  )
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 24.10.0
+    hooks:
+      - id: black
```

Comment:
Just wanted to make somethign sure , so the logic is we want to format the code  with precommit rather than checking for the standars ?? @PublisherName

Reply:
why did i asked the question is when running the pre-commit it fails but it  also format the code using black yml lint and the isort when testing next time it passes so its like magic first it fails and 2nd time it passes due to auto formatiing

Reply:
if it is possible can you  also add the local check for the django check  ``python manage.py check`` duing the precomit  because some changes goes unnoticed sometimes  so the pre-commit can check wether the prtoject can run or not

Reply:
> Just wanted to make somethign sure , so the logic is we want to format the code with precommit rather than checking for the standars ?? @PublisherName

Yeah, exactly @rex9840 ! Running `python -m black .` and `python -m flake8 .` every time we make changes can be confusing, and we might even forget sometimes. With pre-commit configurations, the checks and formatting happen automatically whenever we use the `git commit` command. If any issues are detected, pre-commit takes care of them, ensuring the code meets the standards before it’s committed.

Reply:
> if it is possible can you also add the local check for the django check `python manage.py check` duing the precomit because some changes goes unnoticed sometimes so the pre-commit can check wether the prtoject can run or not

For sure! That’s a great idea. Adding python manage.py check to the pre-commit hooks will help catch any unnoticed issues before committing. Thank you for the suggestion!

Reply:
> why did i asked the question is when running the pre-commit it fails but it also format the code using black yml lint and the isort when testing next time it passes so its like magic first it fails and 2nd time it passes due to auto formatiing

Sorry, my bad! I forgot to fix a trailing whitespace in STYLE_GUIDE.md. I’ve committed the changes.

On the first run, pre-commit tries to fix any automatically solvable errors. If it succeeds, all checks will pass on the second run. But if any issues need manual fixing, they’ll remain until addressed.

Reply:
Can you explain me what happens in case of an error, will the commit be rejected or will the commit be partially accepted ( even in rejection case, the PR will be accepted). 

I generaly like to put it in github actions as it makes it easier for maintainers understand and potentially helps new contributors.

Reply:
If a pre-commit hook fails due to an error, the commit is fully rejected, no part of the commit goes through.

I haven’t configured precommits hooks in github actions for this PR yet, but if you’d like, I can add it to the workflow to automate these checks. Let me know if you'd like me to update this! @n1rjal

Reply:
I think isort is overkill here but I will give it a pass here. If you can remove it and also make a  worflow which will trigger on PR.

- Validation of code styles and pyflake checks on PR must be done during PR.

- We need the workflow to not change the code but only raise an error blocking merge. Later we can support contributors on how to improve it.

Reply:
Thank you for the feedback! I understand the preference for keeping the workflow lightweight, so I'll go ahead and remove **isort** and create a workflow that triggers on PRs. The new workflow will run code style validation and pyflake checks, blocking the merge if any issues are found. It won't make any automatic changes to the code.

I noticed there isn’t a PostgreSQL setup in the CI configuration. Does the Django Checks pass? Or am I missing something? Should I add a PostgreSQL setup to the workflow for the `python manage.py check` to run correctly?

https://github.com/GNOME-Nepal/site-server/blob/674c45f78f7c7dae61b7b9012ddb0835d04ff7e1/.github/workflows/CI.yml#L48-L50

Reply:
```python3
 - name: Run Django system checks 
       run: | 
         python manage.py check || exit 1 
```

Yes, this only checks for syntax errors, and other similar kind of things, it is not a dedicated testing like pytest with some postgresql. So postgresql is not needed here. Thankyou waiting for the PR.